### ISSUE TITLE
Split cxxopts.hpp into multiple files

### ISSUE THREAD

**Issue by eyalroz** [jarro2783/cxxopts#338]
One can observe a rather straightforward decomposition of `cxxopts.hpp` into several parts:

*    non-option-specific string-related code
*    Exception-related code
*    the values namespace and the Value class
*    Options and classes with Option in their name
*    The ParseResult class and some related code.
*    Help-related definitions and code.

And the above is also a topological order of dependencies (except for some implementations which cxxopts.hpp has at the bottom).

So, how about we split the huge single file into a file for each of these? With `cxxopts.hpp` remaining as an includer of all the others?

This will:

* Improve source code readability
* Make it easier to merge changes
* Make the revision history easier to understand (when looking at the affected files)

What do you think, @jarro2783 ?

**Comment by jarro2783** [comment]
I always wanted this to be single header to just drop into a project, and it still doesn't seem huge yet. If there was a nice way to generate a single header then I think it could be broken up.

**Comment by eyalroz** [comment]
@jarro2783 : I'll look into that, but - why is it important to be able to drop a single file rather than a subfolder?

**Comment by jarro2783** [comment]
Yeah I guess having multiple files is not so bad. I'll look at this after some of the other changes because it will mess up PRs.

**Comment by eyalroz** [comment]
So, yeah, the different PRs tend to clash. When you decide which of them you like, I'll rebase them onto each other so that you can merge.

**Comment by lazysegtree** [comment]
What I would like it an automated github CI action that runs on ever push that generates the single header.

I would likely pick it up within Feb.
